### PR TITLE
Use dist.is_init and check torch.is_nccl_available

### LINKED ISSUE #58: torch.distributed.is_mpi_available() + torch.distributed.is_nccl_available()
I am looking at the `setup_ddp() ` function in `utils/utils` - line 78

I don't want to make the code complicated.
However, I think that adding these two checks could be useful to be added in this routine:

- `torch.distributed.is_mpi_available()`
- `torch.distributed.is_nccl_available()`

What do you think?

Please, take a look at the description here:
https://pytorch.org/docs/stable/distributed.html

### PR DESCRIPTION
**Pull request by allaffa**
Fixes #58 (@allaffa if you use these keywords it will automatically close related issues on merge)

### CODE REVIEW

**Review by jychoi-hpc — COMMENTED** [review]
It makes sense. There is also a command to check if gloo is available: `torch.distributed.is_gloo_available()`

File: utils/utils.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def setup_ddp():
 
     """ "Initialize DDP"""
 
-    backend = "nccl" if torch.cuda.is_available() else "gloo"
+    backend = "nccl" if dist.is_nccl_available() else "gloo"
```
Comment (streeve) [inline utils/utils.py:84]:
```suggestion
    backend = "nccl" if dist.is_nccl_available() and torch.cuda.is_available() else "gloo"
```

**Comment by allaffa** [comment]
@streeve @jychoi-hpc 
take look and let me know. Changes are very minor

File: utils/utils.py [hunk 2]
```diff
@@ -56,7 +56,7 @@ def parse_slurm_nodelist(nodelist):
     return nlist
 
 
-def get_comm_size_and_rank():
+def get_raw_comm_size_and_rank():
```
Comment (streeve) [inline utils/utils.py:59]:
```suggestion
def init_comm_size_and_rank():
```

File: utils/utils.py [hunk 3]
```diff
@@ -77,14 +77,31 @@ def get_comm_size_and_rank():
...
+    if dist.is_nccl_available() and torch.cuda.is_available():
+        backend = "nccl"
+    elif torch.distributed.is_gloo_available():
+        backend = "gloo"
+    else:
+        raise RuntimeError("No parallel backends available")
 
-    distributed_data_parallelism = False
-    world_size, world_rank = get_comm_size_and_rank()
+    world_size, world_rank = get_raw_comm_size_and_rank()
```
Comment (streeve) [inline utils/utils.py:104]:
```suggestion
    world_size, world_rank = init_comm_size_and_rank()
```

**Review by streeve — APPROVED** [review]

**Pull request merged**